### DISCUSSION TITLE
STM32 Support

### DISCUSSION THREAD

**Discussion by botoxparty** [davetcc/tcMenu#122] [General]
Hey there,

I love this library however all my embedded projects are using the STM32 platform. I was wondering what the compatibility is like with the STM32 platform?

Particularly I use the STM32 Cube IDE to generate my base project files.

Thanks again

**Comment by davetcc** [comment]
We test against STM32F4 using mbed for every release.

The mbed support works really well, we have one test board using our OLED fork on a network-enabled STM32F439 (working on 1.7) and the other using the LTDC framebuffer and touch screen on an STM32F429 (needs 2.0). Of note, the frame buffer support already includes STM32 Cube files, it brings in the mbed version of the BSP functions.

However, as things stand we have not got around to building an STM Cube only implementation of IoAbstraction and TaskManagerIO but it's on our list. I *think* taskmanagerIO may just work with a few adjustments in the platform determination to use the same settings as mbed (which I think are common ARM CAS constructs). For IoAbstraction, I suspect that a bit more work will be needed around the I2C support and also the GPIO and interrupt abstractions.

As always, we are a very small company, so the problem is always available time against priorities.
